tfutils/base.py

```diff
@@ -358,29 +496,31 @@ def run(sess,
         log.info('Training cancelled since step (%d) is >= num_steps (%d)' % (step, num_steps))
     while step < num_steps:
         old_step = step
+        dbinterface.start_time_step = time.time()
         train_results = sess.run(train_targets)
         step = global_step.eval(session=sess)
         assert (step > old_step), (step, old_step)
         if train_results['loss'] > thres_loss:
             raise HiLossError('Loss {:.2f} exceeded the threshold {:.2f}'.format(train_results['loss'], thres_loss))
-        if step % saver.save_valid_freq == 0 and valid_targets:
-            valid_results = sess.run(valid_targets)
+        if step % dbinterface.save_valid_freq == 0 and valid_targets:
+            valid_results = get_valid_results(sess, valid_targets)
         else:
             valid_results = {}
-        saver.save(train_results, valid_results)
+        dbinterface.save(train_results, valid_results)
     sess.close()
 
 
-def run_base(saver_params,
-             model_params,
-             train_params,
-             loss_params=None,
-             learning_rate_params=None,
-             optimizer_params=None,
-             validation_params=None,
-             thres_loss=100,
-             num_steps=1000000,
-             log_device_placement=False,
+def train_base(save_params,
```

Comment:
I don't like the `_base` anymore – too undescriptive. How about `train_from_params`? Or your suggestion.